### DISCUSSION TITLE
Use of processR in FIMS/tests/concurrent

### DISCUSSION THREAD

**Discussion by kellijohnson-NOAA** [NOAA-FIMS/FIMS#693] [Developers]
I was looking at updating the DESCRIPTION file to update some of the packages that are included. One of them is {processR}. This package is being used in tests/concurrent but it is not listed in the DESCRIPTION file. I was going to add it to Suggests but it is currently not available on CRAN, where it was taken off 2023-02-02 because it requires archived {predict3d}. Is there a different package that we can use instead of {processR}? Because this is in tests/concurrent should I just not worry about adding it to the DESCRIPTION file in suggests? Thanks for the discussion about this 😉. @Bai-Li-NOAA

**Comment by Bai-Li-NOAA** [comment]
@kellijohnson-NOAA, no need to add {processR} to the DESCRIPTION file since it's not included in the {testthat} framework or run in any GHA checks. Tagging @msupernaw to confirm the availability of the [{processR}](https://github.com/nmfs-fish-tools/processR), pretty sure it's a package he developed. 

I’d love to hear more about the goals of these files from @msupernaw and where you all think we should store the concurrent files. To me, they could be turned into a vignette if the idea is to share comparison results with developers or users. Right now, they’re not in any GHA checks since they aren’t written using {testthat}, and we often forget to update them with the latest {FIMS}.

P.S. We do have one {testthat} test that checks if FIMS runs in parallel with {snowfall}, so that’s covered!

**Reply by kellijohnson-NOAA** [reply]
@Bai-Li-NOAA can you please edit this comment to indicate which of the following directories/files inside of "tests" are used in testing, i.e., are ran with `check()` or in GitHub actions?
1. "compilation_tests"
1. "concurrent"
1. "google_benchmark" 
1. "gtest"✅
2. "integration"
3. "test_plan"
1. "testthat" ✅ 
2. "testthat.R" ✅

**Reply by Bai-Li-NOAA** [reply]
updated the comment. We can check with the team to see if anyone wants to keep the files that aren't being used in GHA testing. For example:
- I could move `/google_benchmark` to the collaboration workflow book, and we can bring it back to FIMS when we're ready to write some benchmark tests.
- I can archive `/test_plan` and start using [{testdown}](https://github.com/ThinkR-open/testdown) to auto-generate a higher-level test report.

**Comment by kellijohnson-NOAA** [comment]
[{processR}](https://github.com/nmfs-fish-tools/processR) is not actually on CRAN, instead another package called {processR} is archived on CRAN.
- [ ] @msupernaw is going to rename the package in nmfs-fish-tools

Second, {processR} is beneficial to think about using in FIMS in the future because
* {processR} does not require openMPI, which is a third-party requirement that snowfall and other R packages require.
* {processR} has shared memory to allow read and write at the same time making it where you do not have to transfer data back and forth

**Comment by kellijohnson-NOAA** [comment]
We no longer have any references to processR so I am going to close this Discussion.
